projects/storefrontapp-e2e-cypress/cypress/helpers/quote.ts

```diff
@@ -36,54 +44,86 @@ export function clickOnRequestQuoteInCartAndExpectQuotePage(): void {
  * Clicks on 'Request Quote' on the cart page.
  */
 export function clickOnRequestQuoteInCart(): void {
+  log(
+    'Clicks on "Request Quote" button in the cart',
+    clickOnRequestQuoteInCart.name
+  );
   cy.get('cx-quote-request-button button').click();
 }
-
+/**
+ * Uses a cx-login-form to login a user
+ * @param email Email for the login
+ * @param password Password for the login
+ * @param name Name of the user
+ */
 export function login(email: string, password: string, name: string): void {
-  // Click on the 'Sign in / Register' link
-  // & wait until the login-form is displayed
+  log('Uses a cx-login-form to login a user', login.name);
   cy.get('cx-login [role="link"]')
     .click()
     .then(() => {
       cy.get('cx-login-form').should('be.visible');
     });
-  // Login via authentication service
   authentication.login(email, password);
-  // Verify whether the user logged in successfully,
-  // namely the logged in user should be greeted
   cy.get('.cx-login-greet').should('contain', name);
   cy.get('cx-login').should('not.contain', 'Sign In');
 }
 
+/**
+ * Requests a quote from cart
+ * @param shopName Name of the given shop (Powertools)
+ * @param productName Name of the product that should be used for the quote
+ * @param quantity Quantity of the product used for the quote
+ */
 export function requestQuote(
   shopName,
   productName: string,
   quantity: string
 ): void {
+  log('Requests a quote from cart', requestQuote.name);
   this.addProductToCartForQuotePreparation(shopName, productName, quantity);
   this.clickOnRequestQuoteInCartAndExpectQuotePage();
+  cy.url().as('quoteURL');
 }
 
+/**
+ * Adds a product to the cart
+ * @param shopName Name of the given shop (Powertools)
+ * @param productName Name of the product that should be used for the quote
+ * @param quantity Quantity of the product used for the quote
+ */
 export function addProductToCartForQuotePreparation(
   shopName,
   productName: string,
   quantity: string
 ): void {
+  log('Adds a product to the cart', addProductToCartForQuotePreparation.name);
   common.goToPDPage(shopName, productName);
   this.setQtyOnPD(quantity);
   common.clickOnAddToCartBtnOnPD();
   common.clickOnViewCartBtnOnPD();
 }
 
+/**
+ * Submits a quote via clicking "Yes" button in the confirmation popover
+ */
 export function submitQuote(): void {
+  log(
+    'Submits a quote via clicking "Yes" button in the confirmation popover',
+    submitQuote.name
+  );
   clickOnSubmitQuoteBtnOnQD();
-  clickOnYesBtnOnQuoteSubmitPopUp();
+  clickOnYesBtnWithinRequestPopUp();
+  cy.get<string>('@quoteURL').then(cy.visit);
 }
 
 /**
  * Clicks on 'Submit Quote' on the quote overview page.
  */
 export function clickOnSubmitQuoteBtnOnQD(): void {
```

Comment:
Could you, please, rename this method to `submitQuote()`? Like `requestQuote()`

Reply:
named it into clickSubmitQuote since submitQuote already exists